### PR TITLE
Add segfault handler to PSV test

### PR DESCRIPTION
**Pull request by jystic**
We can now trap segmentation faults and print a stack trace as well as some user data before exiting.

In the case of the PSV test, the user data is the test case which caused the segfault.

### CODE REVIEW

**Comment by jystic** [comment]
This should solve https://github.com/ambiata/icicle/issues/287 more or less

**Comment by tranma** [comment]
oh wow, this is really impressive @jystic

File: data/sea/06-segv.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#define SEGV_USER_DATA_SIZE (1024*1024)
+static char segv_user_data[SEGV_USER_DATA_SIZE];
+
+static void segv_handler (int sig)
+{
+    void *frames[50];
+    size_t n_frames = backtrace (frames, sizeof(frames));
+
+    fprintf (stderr, "icicle: segmentation fault:\n");
+    backtrace_symbols_fd (frames, n_frames, STDERR_FILENO);
```
Comment (amosr) [inline data/sea/06-segv.h]: this is amazing. I had no idea this was possible

File: data/sea/06-segv.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    fprintf (stderr, "\n%s\n", segv_user_data);
```
Comment (tranma) [inline data/sea/06-segv.h]: I wonder what happens when the segv_handler segfaults

File: data/sea/10-mempool.h [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include "05-error.h"
+#include "06-segv.h"
 
 const size_t iblock_size = 1 * 1024 * 1024;
```
Comment (amosr) [inline data/sea/10-mempool.h]: gcc doesn't complain about this one though? maybe it just doesn't like ``static const``?

File: data/sea/52-psv.h [hunk 4]
```diff
@@ -61,11 +61,20 @@ static ierror_msg_t INLINE psv_read_fact
...
 
-/* psv driver */
+
+/*
+Constants
+*/
+
 static const size_t psv_max_row_count   = 128;
 static const size_t psv_buffer_size     = 16*1024;
 static const size_t psv_output_buf_size = 16*1024;
```
Comment (amosr) [inline data/sea/52-psv.h]: hmm never mind

File: data/sea/06-segv.h (hunk 2 above)
Comment (jystic) [inline data/sea/06-segv.h]: o_O probably an infinite loop

Comment (jystic) [inline data/sea/06-segv.h]: I think i'll change this to put the default one back as the first thing it does, then hope that doing that doesn't segfault.

**Comment by amosr** [comment]
great :point_right:

File: data/sea/06-segv.h (hunk 2 above)
Comment (jystic) [inline data/sea/06-segv.h]: segfaults per second
^ yep, this happens

File: data/sea/10-mempool.h (hunk 3 above)
Comment (jystic) [inline data/sea/10-mempool.h]: i think it's because of where it's used rather than the declaration

File: data/sea/06-segv.h (hunk 2 above)
Comment (tranma) [inline data/sea/06-segv.h]: well if this handle segfaults too, probably stack overflow, it's all good :)

Comment (tranma) [inline data/sea/06-segv.h]: ah ok, that's worse!

**Pull request merged**